basis/http/client/client.factor

```diff
@@ -6,17 +6,40 @@ http.parsers io io.crlf io.encodings io.encodings.ascii
 io.encodings.binary io.encodings.iana io.encodings.string
 io.files io.pathnames io.sockets io.timeouts kernel locals math
 math.order math.parser mime.types namespaces present sequences
-splitting urls vocabs.loader ;
+splitting urls vocabs.loader combinators environment ;
 IN: http.client
 
 ERROR: too-many-redirects ;
 
+: success? ( code -- ? ) 200 299 between? ;
+
+ERROR: download-failed response ;
+
+: check-response ( response -- response )
+    dup code>> success? [ download-failed ] unless ;
+
 <PRIVATE
 
+: authority-uri ( request -- str )
+    [ host>> ] [ port>> number>string ] bi ":" glue ;
+
+: absolute-uri ( request -- str )
+    clone f >>username f >>password f >>anchor present ;
+
+: abs-path-uri ( request -- str )
+    relative-url f >>anchor present ;
```

Comment:
This should be a separate commit, I forgot to split it (like the post-data commit). It fixes a bug because the fragment (=anchor) shouldn't be sent to the server. It's a client thing only. For example, curl does this:
```
$ nc -l 3129
$ curl "http://localhost:3129/foo#bar"

GET /foo HTTP/1.1
User-Agent: curl/7.35.0
Host: localhost:3129
Accept: */*
```